src/standalone/Soil/Biogeochemistry/Biogeochemistry.jl

```diff
@@ -141,85 +148,137 @@ SoilCO2Model{FT}(
         drivers::DT,
         toml_dict::CP.ParamDict;
         parameters::SoilCO2ModelParameters{FT} = SoilCO2ModelParameters(toml_dict),
-        boundary_conditions::BC = (
-            top = AtmosCO2StateBC(),
-            bottom = SoilCO2FluxBC((p, t) -> 0.0) # no flux
-        ),
+        boundary_conditions = nothing,
         sources::Tuple = (MicrobeProduction{FT}(),),
-    ) where {FT, BC, DT}
+    ) where {FT, DT}
 
 A constructor for `SoilCO2Model`.
 Defaults are provided for the parameters, boundary conditions, and sources.
 These can be overridden by providing the appropriate keyword arguments.
+Boundary conditions are nested: top/bottom each have separate `co2` and `o2` BCs.
+
+If `boundary_conditions` is not provided (or set to `nothing`), the default boundary conditions are:
+- Top: AtmosCO2StateBC() for CO2, AtmosO2StateBC(earth_param_set, M_O2, O2_a) for O2
+- Bottom: Zero flux for both CO2 and O2
 """
 function SoilCO2Model{FT}(
     domain::ClimaLand.AbstractDomain,
     drivers::DT,
     toml_dict::CP.ParamDict;
     parameters::SoilCO2ModelParameters{FT} = SoilCO2ModelParameters(toml_dict),
-    boundary_conditions::BC = (
-        top = AtmosCO2StateBC(),
-        bottom = SoilCO2FluxBC((p, t) -> 0.0), # no flux
-    ),
+    boundary_conditions = nothing,
     sources::Tuple = (MicrobeProduction{FT}(),),
-) where {FT, BC, DT}
+) where {FT, DT}
+    # Set default boundary conditions if not provided
+    if boundary_conditions === nothing
```

Comment:
What if we made this the default, instead of `nothing`? I think it achieves the same purpose
```
parameters::SoilCO2ModelParameters{FT} = SoilCO2ModelParameters(toml_dict),
boundary_conditions = (
            top = (
                co2 = AtmosCO2StateBC(),
                o2 = AtmosO2StateBC(
                    parameters.earth_param_set,
                    parameters.M_O2,
                    parameters.O2_a,
                ),
            ),
            bottom = (
                co2 = SoilCO2FluxBC((p, t) -> 0.0),
                o2 = SoilCO2FluxBC((p, t) -> 0.0),
            ),
        )
```